go.mod

```diff
@@ -184,3 +184,18 @@ require (
 	tags.cncf.io/container-device-interface v0.8.0 // indirect
 	tags.cncf.io/container-device-interface/specs-go v0.8.0 // indirect
 )
+
+exclude (
+	// These dependencies were updated to "master" in some modules we depend on,
```

Comment:
what modules? (I guess nydus-snapshotter and stargz-snapshotter but not sure)

Reply:
If it is already in containerd then what is it breaking for "all users of the containerd module" ? Or doesn't this need to go to containerd's go.mod then?

Reply:
The new versions won't break things as there are no code changes. But  they'll never be released because both projects are archived, so these  should  never have gotten in any project  as dependency.  Getting rid of those untagged  versions is the tricky bit, due to how we have many (circular) dependencies, so I'm trying to cut  that circle.

Once buildkit enforces it, now it will be enforced by buildx and moby, and thus  docker/cli, which then forces buildkit to use the version.

Reply:
s/archived/ or marked "unmaintained"

Reply:
Are these excludes more "dependency hygiene" than fixing an actual issue? I'm trying to figure out if we are going to keep these excludes "forever".

Reply:
(Assuming exclude propagates but gets overwritten by other deps)

Can we add a comment that these are downgraded in containerd 2.0.2 but because `nydus-snapshotter`, `stargz-snapshotter` do not depend on 2.0.2 yet that downgrade does not work unless we make one explicitly.

Reply:
There is no partucular issue. The world still exists, the wheels keep rolling.

I look at go.mod and see a noticeable amount of untagged deps. But for some reason, @thaJeztah cares a lot more about these two than about others.

> I'm trying to figure out if we are going to keep these excludes "forever".

@thaJeztah hopes he can stop the plague by quick downgrade of every place where these deps already got (though nothing prevents this from happening again).

Reply:
The other untagged versions don't  tag releases, so there's no release  to choose   from.